src/java/org/apache/cassandra/service/reads/repair/RowIteratorMergeListener.java

```diff
@@ -280,7 +280,8 @@ public void onMergedRangeTombstoneMarkers(RangeTombstoneMarker merged, RangeTomb
                 // current deletion, this means the current deletion is due to a previously open range tombstone,
                 // and if the source isn't currently repaired for that RT, then it means it's up to date on it).
                 DeletionTime partitionRepairDeletion = partitionLevelRepairDeletion(i);
-                if (markerToRepair[i] == null && currentDeletion.supersedes(partitionRepairDeletion))
+
+                if (markerToRepair[i] == null && !currentDeletion.supersedes(partitionRepairDeletion))
```

Comment:
Are we sure about the change on this line? I *think* the condition swap is incorrect, or at the very least the comment above it now is.